add key binding to copy code

Sneezry opened the pull request:
Press arrow keys to select a code, press `enter` to copy.

![Animation](https://user-images.githubusercontent.com/1621293/123319110-90560580-d562-11eb-9889-96240cf1cab6.gif)

Sneezry committed:
add key binding to copy code

mymindstorm commented on src/components/Popup/MainBody.vue:
```diff
@@ -90,6 +90,87 @@ export default Vue.extend({
+    },
+    getTabindex(
+      entry: OTPEntry,
+      entries: OTPEntry[],
```
entries, shouldFilter, and filter are all computed properties and don't need to be passed as arguments.

mymindstorm commented on src/components/Popup/MainBody.vue:
```diff
@@ -90,6 +90,87 @@ export default Vue.extend({
+    },
+    findNextEntryIndex(
+      entries: OTPEntry[],
+      shouldFilter: boolean,
```
same as above.

mymindstorm commented on src/components/Popup/MainBody.vue:
```diff
@@ -90,6 +90,87 @@ export default Vue.extend({
+      return entry === firstEntry ? 0 : -1;
+    },
+    findNextEntryIndex(
+      entries: OTPEntry[],
```
On all functions, entries, shouldFilter, and filter are all computed properties and don't need to be passed as arguments.

Sneezry replied:
Updated.

mymindstorm commented:
I really like this!

Bugs / Other:
- When entering password, it does not function. Need to hit tab or reopen the extension for it to work.
- If I click anywhere else (even an entry) it will not let me use arrow keys until tab or reopen. 
	- Should we listen for keys on the document?
	- Should we focus entries on click?

Sneezry committed:
address comment

Sneezry commented:
> When entering password, it does not function. Need to hit tab or reopen the extension for it to work.

Fixed. After applying password, the first entry will be auto focused.

> If I click anywhere else (even an entry) it will not let me use arrow keys until tab or reopen.

Entry will have focus when it is clicked. If you click anywhere else, the entry should not have focus. And is as designed.

> Should we listen for keys on the document?

I don't think so. By following w3 accessibility guide, `tab` should be used to select active element, and arrow keys should be used to select items in an active group element. If the group element is not active, we should not response to the action.

> Should we focus entries on click?

Yes. This is a bug in previous commit. I have fixed that in the recent commit.

mymindstorm commented on src/components/Popup/MainBody.vue:
```diff
@@ -14,6 +14,7 @@
         v-model="searchText"
         v-bind:placeholder="i18n.search"
         type="text"
+        tabindex="-1"
```
I think this should be able to be tabbed to, but it is ok to not allow it until we get around to overhaling keyboard nav for the header and menu. Hopefully compoentizing everything will help at the second version of that.

mymindstorm reviewed (approved):
Looks great

mymindstorm commented:
Will this supersede #689?

Sneezry commented:
> Will this supersede #689?

I suppose. Number keys could be hard to use to copy a specific entry. Imaging how to copy the 7th entry in the list. I have to count entries from top to bottom.
